### PR TITLE
Create elastic-agent-pull-requests.yml

### PR DESCRIPTION
**Pull request by jlind23**
Automatically move PR to an Elastic Agent board as soon as they are labelled with either "team:elastic-agent-control-plane" or "team:elastic-agent-data-plane".

### CODE REVIEW

**Review by cmacknz — APPROVED** [review]

**Review by ph — APPROVED** [review]
Not completely familiar with github action but the graphql looks good.

**Comment by jlind23** [comment]
@v1v this beats-ci/pr-merge seems to fail because there is no changes made to the code base. Shouldn't we exclude some repositories from the check?

**Comment by jlind23** [comment]
@ph if you want give a try at a graphql request, github offers this: https://graphql.github.com/

**Review by ruflin — APPROVED** [review]

**Comment by ruflin** [comment]
@jlind23 I wonder why the action to add it to the Data plane board was skipped?

File: .github/workflows/elastic-agent-pull-requests.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Add to Elastic Agent Data Plane or Control Plane Board
```
Comment (ruflin) [inline .github/workflows/elastic-agent-pull-requests.yml:1]: Nit: If you shorten the title, this looks more descriptive.

Screenshot 2022-02-04 at 09 35 40

**Comment by jlind23** [comment]
@ruflin it was not. Each labelling generate an event on github side which then generate a github action run.
2 labels means two runs. If you check each one of those you will find both condition being succesfull.

File: .github/workflows/elastic-agent-pull-requests.yml (hunk 1 above)
Comment (jlind23) [reply .github/workflows/elastic-agent-pull-requests.yml:1]: Yup it keeps only the last run in it.

**Comment by ruflin** [comment]
Triggering the tests again as it seems it was aborted (manually?)

**Comment by v1v** [comment]
> @v1v this beats-ci/pr-merge seems to fail because there is no changes made to the code base. Shouldn't we exclude some repositories from the check?


@jlind23 , as far as I see there were two consecutive commits and the on-going builds were killed for the new one. That's the existing approach since the very beginning.

If no code changes, then only the linting is validated, we could exclude the .github folder, but for simplicity I'd rather prefer to run a basic linting than adding more complexity in the CI.

What do you think?

**Comment by jlind23** [comment]
@v1v basic linting suits me well then. I am for reducing the complexity too.

**Comment by cmacknz** [comment]
I think this is causing some CI failures because `GITHUB_TOKEN` is missing, example: https://github.com/elastic/beats/runs/5098115833?check_suite_focus=true

**Pull request merged**